### PR TITLE
[podman run] --rm option shold conflicts with --restart

### PR DESCRIPTION
**Pull request by zhangguanzhang**
FIxes: https://github.com/containers/podman/issues/7878
Signed-off-by: zhangguanzhang <zhangguanzhang@qq.com>

### CODE REVIEW

File: cmd/podman/common/specgen.go [hunk 1]
```diff
@@ -233,7 +233,7 @@ func FillOutSpecGen(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string
 
 	// validate flags as needed
 	if err := c.validate(); err != nil {
-		return nil
+		return err
```
Comment (zhangguanzhang) [inline cmd/podman/common/specgen.go:236]: resaon is in the line

**Comment by rhatdan** [comment]
LGTM

**Comment by zhangguanzhang** [comment]
maybe this step need to restart
```
test ubuntu-19 RCLI:false
```

**Comment by mheon** [comment]
LGTM once this goes green

**Pull request merged**